### ISSUE TITLE
accumulate: test failure message prints unhelpful function pointer, go vet complains

### ISSUE THREAD

**Issue by petertseng** [exercism/xgo#317]
I know there was a past work on this in https://github.com/exercism/exercism.io/pull/1260 , but now `go vet` has something to say again:

```
accumulate_test.go:33: arg test.converter in printf call is a function value, not a function call
```

Here is an example of the error message you will see if your implementation fails a test:

```
--- FAIL: TestAccumulate (0.00s)
	accumulate_test.go:35: PASS: echo []
	accumulate_test.go:35: PASS: echo [echo echo echo echo]
	accumulate_test.go:33: Accumulate([first letter only], (func(string) string)(0x47de00)): expected [First Letter Only], actual [first letter only]
```

That `(func(string) string)(0x47de00)` is decidedly unhelpful. I can think of two possibilities to replace:

1. The name of the function - however that appears to need reflection, unless I am mistaken.
2. The description (just like the PASS message does)

**Comment by kytrinyx** [comment]
I am OK with introducing reflection if it helps give the solver better hints, but as a first pass, the description is probably fine.

**Comment by petertseng** [comment]
Adding good first patch. If anyone is out there, hello, we would be very pleased if you would like to do this. You will need to edit [exercises/accumulate/accumulate_test.go](https://github.com/exercism/xgo/blob/master/exercises/accumulate/accumulate_test.go) to make the error message better.

You can test your change by intentionally making a failing implementation: For instance, try taking [the example implementation](https://github.com/exercism/xgo/blob/master/exercises/accumulate/example.go) and making it just return an empty array, or return the input array unmodified. Then run `go test`.

You should see the error message similar to the one I pasted above, and what we're looking for is that something helpful is printed to describe the function being passed to `Accumulate`, rather than something like `(func(string) string)(0x47de00)`

**Comment by petertseng** [comment]
Closed by #337, many thanks.

**Comment by petertseng** [comment]
I have just found out today - this track deprecated accumulate at 8672c43c9207c7d738db173b066d16dcd968b381, so nobody was even seeing the unclear message!

I do not know why it was deprecated.
